recipe/meta.yaml

```diff
@@ -1,14 +1,13 @@
-{% set name = "dask-ml" %}
-{% set version = "2023.3.24" %}
+{% set name = "dask_ml" %}
```

Comment:
Looks like the name is still `dask-ml`, see https://github.com/dask/dask-ml/blob/v2024.4.4/pyproject.toml#L6